crates/ruff_linter/src/rules/flake8_bugbear/rules/duplicate_exceptions.rs

```diff
@@ -209,14 +209,14 @@ pub(crate) fn duplicate_exceptions(checker: &mut Checker, handlers: &[ExceptHand
         }
     }
 
-    let is_star = checker
-        .semantic()
-        .current_statement()
-        .as_try_stmt()
-        .is_some_and(|try_stmt| try_stmt.is_star);
     if checker.enabled(Rule::DuplicateTryBlockException) {
         for (name, exprs) in duplicates {
             for expr in exprs {
+                let is_star = checker
+                    .semantic()
+                    .current_statement()
+                    .as_try_stmt()
+                    .is_some_and(|try_stmt| try_stmt.is_star);
```

Comment:
Just for learning purposes. @MichaReiser 
Why was this moved inside the `diagnostic-branch`?

I thought calculating it outside the loop once would be better.